main.c

```diff
@@ -190,15 +203,19 @@ singlethreaded(struct task_t *task, struct config_t *config)
     struct st_context_t context;
     context.hash = config->hash;
     context.cd.initialized = 0;
-    bool found = false;
+    context.config = config;
+
+    task->from = 0;
+    task->to = strlen(config->alphabet) - 1;
```

Comment:
'to' is a position in password template. It has nothing to do with length of the alphabet